### PR TITLE
Core: Fix compatibility with sinon.useFakeTimers in IE

### PR DESCRIPTION
**Pull request by timmywil**
The Promise polyfill needed to save the unmodified setImmediate function in IE to use for Promise._immediateFn

### CODE REVIEW

File: test/main/promise.js [hunk 1]
```diff
@@ -247,4 +265,25 @@ QUnit.module('Support for Promise', function () {
...
+      this.setTimeout = globalNS.setTimeout;
+      globalNS.setTimeout = function () {};
+      this.setImmediate = globalNS.setImmediate;
+      globalNS.setImmediate = function () {};
+      // Adds 1 assertion
+      return createMockPromise(assert);
+    },
+    afterEach: function (assert) {
+      globalNS.setTimeout = this.setTimeout;
+      globalNS.setImmediate = this.setImmediate;
```
Comment (Krinkle) [inline test/main/promise.js:280]: This leaves behind an undefined global property in environments where the property did not exist. Can we avoid that?

File: lib/promise-polyfill.js [hunk 2]
```diff
@@ -373,16 +373,18 @@ Promise.race = function(arr) {
 };
 
 // Use polyfill for setImmediate for performance gains
-Promise._immediateFn =
+// @ts-ignore
```
Comment (Krinkle) [inline lib/promise-polyfill.js:376]: Please update the `Patches for use in QUnit:` header to document this local patch, to make sure we remember it in the future :)

**Review by Krinkle — CHANGES_REQUESTED** [review]
Two inline issues commented.

**Review by Krinkle — APPROVED** [review]

**Comment by Krinkle** [comment]
Looks like GitHub got itself in some kind of broken state, not agreeing on which commit's status to look at for the merge button, thus blocking me from merging due to waiting for a status that would never come. Closing and re-opening fixed this.

**Comment by timmywil** [comment]
Thanks! If you could do a release, I can update the jQuery branch.

**Review by Dearley78 — COMMENTED** [review]
lib/promise-polyfill.js

**Pull request merged**